crates/aptos-dkg/src/range_proof.rs

```diff
@@ -465,23 +495,37 @@ fn byte_to_bits_le(val: u8) -> Vec<bool> {
 
 /// Compute alpha, beta.
 fn fiat_shamir_challenges(
-    _com: &Commitment,
+    com: &Commitment,
     c: &[G1Projective],
     c_hat: &[G2Projective],
-    with_alphas: bool,
+    with_alphas: bool, // atm this has to be true for both prover and verifier, because FS transcripts have to be consistent
+    fs_transcript: &mut merlin::Transcript,
 ) -> (Vec<Scalar>, Vec<Scalar>) {
     assert_eq!(c.len(), c_hat.len());
-    //TODO: real fiat-shamir.
-    let mut rng = StdRng::from_seed([0; 32]);
-    let beta_vals = (0..c.len())
-        .map(|_| random_128bit_scalar(&mut rng))
-        .collect();
+
+    <merlin::Transcript as fiat_shamir::RangeProof<Scalar>>::append_range_proof_sep(fs_transcript);
+
+    // Add the first prover message to the transcript
+    <merlin::Transcript as fiat_shamir::RangeProof<Scalar>>::append_range_proof_claim(
+        fs_transcript,
+        &(com, c),
+    );
+
+    // Generate the Fiat–Shamir challenges from the updated transcript
+    let beta_vals =
+        <merlin::Transcript as fiat_shamir::RangeProof<Scalar>>::challenge_linear_combination_128bit(
+            fs_transcript,
+            c.len(),
+        );
+
     let alpha_vals = if with_alphas {
-        (0..c.len())
-            .map(|_| random_128bit_scalar(&mut rng))
-            .collect()
+        // TODO: get rid of this, with_alpha is always true
```

Comment:
I thought you rightfully pointed out we should skip over generating alphas during proving, so makes sense to keep it, no?

Don't matter in the long term, cause [the new DeKART version](https://alinush.github.io/dekart#mathsfdekartmathsffftmathsfprovemathcalfscdotmathsfprk-c-ell-z_0ldotsz_n-1-rrightarrow-pi) doesn't have $\alpha$'s.

Reply:
> all of the prover's challenges should be derived by the verifier separately in any kind of protocol, but did not get around to implementing that cleanly.

That sounds like you're re-describing Fiat-Shamir, no? Let's discuss. Not sure what you mean.